common.sh

```diff
@@ -256,8 +273,9 @@ test_ssh()
         ssh -T -o ConnectTimeout=30 -o StrictHostKeyChecking=no -o BatchMode=yes -i ~/${slave_keypair} ${ami[1]}@$1  hostname
         if [ $? -eq 0 ]; then
             slave_ready=0
-            echo "SSH connection of instance $1 is ready"
             set -xe
+            ssh -T -o ConnectTimeout=30 -o StrictHostKeyChecking=no -o BatchMode=yes -i ~/${slave_keypair} ${ami[1]}@$1 "$(typeset -f); boot_finished_check"
```

Comment:
Does it work? We can call function defined in local bash script on remote instance?

Reply:
Yeah, I did some basic testing of this to verify. Typeset lets you use functions you've defined locally.